vlog4j-client/src/test/java/org/vlog4j/client/picocli/PrintQueryResultsTest.java

```diff
@@ -36,27 +42,94 @@ public void validate_completeFalseSizeTrue_valid() throws ConfigurationException
 	}
 
 	@Test
-	public void validate_completeTrue_valid() throws ConfigurationException {
+	public void validate_sizeFalseCompleteTrue_valid() throws ConfigurationException {
 		PrintQueryResults prq = new PrintQueryResults();
 		prq.setSizeOnly(false);
 		prq.setComplete(true);
 		prq.validate();
 	}
 
 	@Test(expected = ConfigurationException.class)
-	public void validate_completeTrueSizeTrue_notValid() throws ConfigurationException {
+	public void validate_sizeTrueCompleteTrue_notValid() throws ConfigurationException {
 		PrintQueryResults prq = new PrintQueryResults();
 		prq.setSizeOnly(true);
 		prq.setComplete(true);
 		prq.validate();
 	}
 
 	@Test
-	public void validate_completeFalseSizeFalse_valid() throws ConfigurationException {
+	public void validate_sizeFalseCompleteFalse_valid() throws ConfigurationException {
 		PrintQueryResults prq = new PrintQueryResults();
 		prq.setSizeOnly(false);
 		prq.setComplete(false);
 		prq.validate();
 	}
 
+	@Test
+	public void printConfiguration_sizeTrueCompleteFalse_valid() {
+		PrintQueryResults prq = new PrintQueryResults();
+		prq.setSizeOnly(true);
+		prq.setComplete(false);
+		assertEquals(String.format(outputConfigurationBase, true, false), captureOutputPrintConfiguration(prq));
+	}
+
+	@Test
+	public void printConfiguration_sizeFalseCompleteTrue_valid() {
+		PrintQueryResults prq = new PrintQueryResults();
+		prq.setSizeOnly(false);
+		prq.setComplete(true);
+		assertEquals(String.format(outputConfigurationBase, false, true), captureOutputPrintConfiguration(prq));
+	}
+
+	@Test
+	public void printConfiguration_sizeTrueCompleteTrue_notValid() {
+		PrintQueryResults prq = new PrintQueryResults();
+		prq.setSizeOnly(true);
+		prq.setComplete(true);
+		assertEquals(String.format(outputConfigurationBase, true, true), captureOutputPrintConfiguration(prq));
+	}
+
+	@Test
+	public void printConfiguration_sizeFalseCompleteFalse_valid() {
+		PrintQueryResults prq = new PrintQueryResults();
+		prq.setSizeOnly(false);
+		prq.setComplete(false);
+		assertEquals(String.format(outputConfigurationBase, false, false), captureOutputPrintConfiguration(prq));
+	}
+
+	@Test
+	public void setSizeOnly_and_isSizeOnly() {
+		PrintQueryResults prq = new PrintQueryResults();
+		prq.setSizeOnly(false);
+		assertEquals(false, prq.isSizeOnly());
+		prq.setSizeOnly(true);
+		assertEquals(true, prq.isSizeOnly());
+	}
+
+	@Test
+	public void setComplete_and_isComplete() {
+		PrintQueryResults prq = new PrintQueryResults();
+		prq.setComplete(false);
+		assertEquals(false, prq.isComplete());
+		prq.setComplete(true);
+		assertEquals(true, prq.isComplete());
+	}
+
+	private String captureOutputPrintConfiguration(PrintQueryResults prq) {
```

Comment:
Mockito can perhaps be used to simulate actually printing to System.out, or the output stream to print to can be a method parameter